src/secret_sharing/replicated/semi_honest/xor_share.rs

```diff
@@ -73,47 +78,33 @@ impl<V: BooleanShare> BitXorAssign<&Self> for XorShare<V> {
     }
 }
 
-impl<V: BooleanShare> Serializable for XorShare<V> {
-    const SIZE_IN_BYTES: usize = 2 * V::SIZE_IN_BYTES;
-
-    fn serialize(self, buf: &mut [u8]) -> io::Result<()> {
-        if buf.len() < Self::SIZE_IN_BYTES {
-            Err(io::Error::new(
-                io::ErrorKind::WriteZero,
-                "not enough buffer capacity",
-            ))
-        } else {
-            let (left, right) = (self.left(), self.right());
-            left.serialize(&mut buf[..V::SIZE_IN_BYTES])?;
-            right.serialize(&mut buf[V::SIZE_IN_BYTES..2 * V::SIZE_IN_BYTES])?;
-
-            Ok(())
-        }
-    }
-
-    fn deserialize(buf: &[u8]) -> io::Result<Self> {
-        if buf.len() < Self::SIZE_IN_BYTES {
-            Err(io::Error::new(
-                io::ErrorKind::UnexpectedEof,
-                "not enough buffer capacity",
-            ))
-        } else {
-            let left = V::deserialize(&buf[..V::SIZE_IN_BYTES])?;
-            let right = V::deserialize(&buf[V::SIZE_IN_BYTES..2 * V::SIZE_IN_BYTES])?;
-
-            Ok(Self::new(left, right))
-        }
+impl<V: BooleanShare> Serializable for XorShare<V>
+where
+    V::Size: Add<V::Size>,
+    <V::Size as Add<V::Size>>::Output: ArrayLength<u8>,
+{
+    type Size = <V::Size as Add<V::Size>>::Output;
```

Comment:
Clever.  typenum is completely bonkers.  The use of Add to generate derived types seems like a misuse of the trait; but this is fine.